auditlog/models.py

```diff
@@ -486,6 +493,31 @@ def changes_display_dict(self):
             changes_display_dict[verbose_name] = values_display
         return changes_display_dict
 
+    def _get_changes_display_for_fk_field(
+        self, field: Union[models.ForeignKey, models.OneToOneField], value: Any
+    ) -> str:
+        """
+        :return: An object display string for a given FK value and the field to which it belongs
+        """
+        # Return "None" if the FK value is "None".
+        if value == "None":
+            return value
+
+        # Attempt to convert given value to the PK type for the related model
+        try:
+            pk_value = field.related_model._meta.pk.to_python(value)
+        # ValidationError will handle legacy values where string representations were
+        # stored rather than PKs. This will also handle cases where the PK type is
+        # changed between the time the LogEntry is created and this method is called.
+        except ValidationError:
+            return value
+        # Attempt to return the string representation of the object
+        try:
+            return smart_str(field.related_model.objects.get(pk=pk_value))
+        # ObjectDoesNotExist will be raised if the object was deleted.
+        except ObjectDoesNotExist:
+            return f"Deleted '{field.related_model.__name__}' ({value})"
+
```

Comment:
One consequence of this is that the related objects' str representation is based on the CURRENT state of that object, not the historical state. This is not... great when looking at the audit log of object A, it should not tell you the lifecycle of object B. 

There are probably ways around this by using the serialized data feature, but this has quite a few drawbacks as serialized data may not be compatible with the current code. 

One possible solution is storing a tuple in the changes dict `(pk, str)` but this changes the shape in other ways and represents a more complex breaking change. 

**or maybe having a datetime in a __str__ method is just an antipattern** (the root issue, on why we are here)

Reply:
String representations are usually used as a way to have a human readable display for the object. I do not believe that `__str__` is intended to show state. Therefore showing the human readable form of which object was attached to this field on this object may not be so problematic. One could also argue that displaying the current version of the string representation for both the old and new relationships is a feature add because it's easier to compare. That said, I don't think people are likely to change the format of the returned `__str__` value that often. (Although losing the string representation when the object is deleted is unfortunate.)